src/init.cpp

```diff
@@ -681,6 +681,7 @@ void ThreadImport(std::vector<fs::path> vImportFiles)
         LoadMempool();
         fDumpMempoolLater = !fRequestShutdown;
     }
+    g_is_mempool_loaded = true;
```

Comment:
It would be nice to drop fDumpMempoolLater and only have one global for "mempool was loaded". Should be able to set g_is_mempool_loaded to !fRequestShutdown here, and then check -persistmempool before flushing during shutdown to avoid changing behavior without adding another boolean.

Reply:
Perhaps I'm not understanding the code, but what I assumed is that fDumpMempoolLater is set to !fRequestShutdown instead of true to avoid more than one thread dumping the mempool. But in this case we want all threads to work with savemempool once the mempool is loaded.
Am I confused about this or missing something?

Reply:
There is only one shutdown thread, not sure what you mean "more than one thread dumping the mempool". !fRequestShutdown is a check to make sure LoadMempool() didn't quit early, but its also safe to re-use that in RPC as if fRequestShutdown is set we'll be shutting down soon anyway, so RPCs can fail.

Reply:
Alright, I wasn't understanding why !fRequestShutdown was set, it seems we can have a single boolean then and drop fDumpMempoolLater.